hooks/pre-push.d/rs-spdx-license-test

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -e
+top=`git rev-parse --show-toplevel`
+status=0
+for file in $(find "$top" -path "$top"/target -prune -o -name "*.rs")
```

Comment:
```suggestion
for file in $(find "$top" -type f -name "*.rs")
```

Reply:
That line excludes the locally-generated `target` directory, which includes a lot of generated `.rs` files without a license. The test fails if it's removed, since those generated files get tested for license headers they don't have.

Reply:
@mbestavros Thanks, I misunderstood that part. Yours is obviously better.

Reply:
@mbestavros You still probably want `-type f` that will remove the need for your `[ -f $file ]` check later.

Reply:
@npmccallum That's, again, related... This `find` invocation returns the right `.rs` files, but it also returns... the `target` directory itself, for some reason. I've already tried specifying `-type f`, but it doesn't seem to do anything -- the directory is still returned as a result.

Checking for `[ -f "$file" ]` later works. With my implementation it was necessary, since `sed` would fail when trying to read the `target` directory. Although, having tried your suggested `awk` line instead of `sed`, the test works without the `[ -f "$file" ]` check -- but it produces an error message.

If we can't fix the `target` directory in the `find` invocation, I prefer having `[ -f "$file" ]` to make the output clean.